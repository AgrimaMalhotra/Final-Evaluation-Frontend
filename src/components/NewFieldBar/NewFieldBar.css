.new-field-bar {
  display: flex;
  width: 100%;
  height: 48px;
  background-color: white;
  border-radius: 10px;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  font-weight: 500;
  align-items: center;
  margin: 10px;
}

.field-type-logo {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.2rem;
  width: 5%;
  height: 100%;
  background-color:rgba(0, 0, 255, 0.588);
  color: white;
  border-top-left-radius: 10px;
  border-bottom-left-radius: 10px;
}

.field-name {
  margin-left: 14px;
  font-size: 1.1rem;
  width: 36%;
}

.field-type-text{
  font-size: 1.1rem;
  color: rgba(0, 0, 0, 0.429);
  width: 50%;
}

.action-buttons {
  display: flex;
  width: 5%;
  justify-content: space-between;
}