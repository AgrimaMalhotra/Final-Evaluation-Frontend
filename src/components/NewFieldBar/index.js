import * as React from 'react';
import './NewFieldBar.css';
import editIcon from '../../assets/user-edit-text-message-note_2023-03-09/user-edit-text-message-note.png';
import deleteIcon from '../../assets/trash-delete-recycle-bin-bucket-waste_2023-03-09/trash-delete-recycle-bin-bucket-waste.png';
import propTypes from 'prop-types';

const NewFieldBar = ( {fields}) => {
  const icon={
    'text': 'Ab',
    'string': 'Ab',
    'number': '123',
    'date': 'D',
  };
  return (
    <div className="new-field-bar">
      <div className='field-type-logo'>
        <p>{icon[fields[1]]}</p>
      </div>
      <p className='field-name'>{fields[0]}</p>
      <p className='field-type-text'>{fields[1]}</p>
      <div className='action-buttons'>
        <img src={editIcon}></img>
        <img src={deleteIcon}></img>
      </div>

    </div>
  );
};

NewFieldBar.propTypes = {
  fields: propTypes.array.isRequired,
};
export default NewFieldBar;