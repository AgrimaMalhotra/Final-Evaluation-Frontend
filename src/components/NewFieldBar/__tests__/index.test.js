import React from 'react';
import {render} from '@testing-library/react';
import NewFieldBar from '../index';

describe('NewFieldBar', () => {
  const mockFields = 
    {
      'id': 'number',
      'name': 'text',
    };
  it('should render correctly', () => {
    const component = render(<NewFieldBar fields={mockFields}/>);
    expect(component).toMatchSnapshot();
  });
});
