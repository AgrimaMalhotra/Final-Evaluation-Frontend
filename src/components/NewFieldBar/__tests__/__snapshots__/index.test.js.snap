// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewFieldBar should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="new-field-bar"
      >
        <div
          class="field-type-logo"
        >
          <p />
        </div>
        <p
          class="field-name"
        />
        <p
          class="field-type-text"
        />
        <div
          class="action-buttons"
        >
          <img
            src="user-edit-text-message-note.png"
          />
          <img
            src="trash-delete-recycle-bin-bucket-waste.png"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="new-field-bar"
    >
      <div
        class="field-type-logo"
      >
        <p />
      </div>
      <p
        class="field-name"
      />
      <p
        class="field-type-text"
      />
      <div
        class="action-buttons"
      >
        <img
          src="user-edit-text-message-note.png"
        />
        <img
          src="trash-delete-recycle-bin-bucket-waste.png"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
